src/main/java/duke/parsers/InputParser.java

```diff
@@ -0,0 +1,205 @@
+            }
+        }
+        //create deadline object, splitting the due date by "/" and stripping off the by:
+        else if (splitStr[0].equals("deadline")) {
+            try {
+                inputChecker(splitStr, "deadline");
+                String[] deadlineArr = input.split("/by ");
+                Date deadline = dateParser(deadlineArr[1]);
+
+                Deadline d = new Deadline(deadlineArr[0].substring(9), deadline);
+                tasks.add(d);
+                if (toStore){
+                    Storage.saveTasks(tasks);
+                };
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+        //create event object, splitting the due date by "/" and stripping off the to: and from:
+        else if (splitStr[0].equals("event")) {
+            try {
+                inputChecker(splitStr, "event");
+                int startIndex = input.indexOf("/from ");
+                int endIndex = input.indexOf("/to");
+
+                Date from = dateParser(input.substring(startIndex + 6, endIndex - 1));
+                Date to = dateParser(input.substring(endIndex + 4));
+
+                Event e = new Event(input.substring(6, startIndex), from, to);
+                tasks.add(e);
+                if (toStore){
+                    Storage.saveTasks(tasks);
+                };
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+
+        //find a certain task
+        else if (splitStr[0].equals("find")){
+            try{
+                inputChecker(splitStr, "find");
+                String toFind = input.substring(5);
+                ArrayList<Task> foundTasks = new ArrayList<>();
+
+                for (Task task:tasks){
+                    if (task.getDescription().contains(toFind)){
+                        foundTasks.add(task);
+                    }
+                }
+                if (foundTasks.size() >0){
+                    System.out.println("Here are the matching tasks in your list:");
+                    for (int i = 0; i < foundTasks.size(); i++) {
+                        int index = i + 1;
+                        System.out.println("  " + index + "." + foundTasks.get(i).toString());
+                    }
+                } else{
+                    System.out.println("No tasks called " + toFind + " found");
+                }
+            } catch (DukeException e){
+                System.out.println(e.getMessage());
+            }
+        }
+
+        //mark or unmark an existing task
+        else if (splitStr[0].equals("mark") || splitStr[0].equals("unmark")) {
+            try {
+                inputChecker(splitStr, "mark/unmark");
+                int index = Integer.parseInt(splitStr[1]);
+                Task item = tasks.get(index - 1);
+                item.setAction(splitStr[0]);
+                if (toStore){  Storage.saveTasks(tasks); };
```

Comment:
Conditionals should be put in a separate line as per coding standards.
```suggestion
                if (toStore) {
                    Storage.saveTasks(tasks); 
                };
```